ckanext/data_qld/controller_functions.py

```diff
@@ -75,3 +81,368 @@ def show_schema(dataset_id, resource_id):
     except tk.NotAuthorized as e:
         log.warn(e)
         return tk.abort(403, _('You are not authorized to view the Data Scheme for the resource %s' % resource_id))
+
+
+def _valid_report_type(report_type):
```

Comment:
can you put the 'report' controller functions into a new class to seperate out and give context that this is 'extensions' to the report plugin etc.

Reply:
.. seems your merged them, why is that?

Reply:
Hadn't seen this comment, but yes, I can split them.